docs/NooBaaNonContainerized/Upgrade.md

```diff
@@ -69,5 +69,161 @@ The currently available upgrade process of NooBaa Non Containerized is an offlin
     cat /etc/noobaa.conf.d/system.json
     {"hostname":{"current_version":"5.17.0","upgrade_history":{"successful_upgrades":[{"timestamp":1719299738760,"completed_scripts":[],"from_version":"5.15.4","to_version":"5.17.0"}]}}}
     ```
-## Online Upgrade
-The process of Online Upgrade of Non Containerized NooBaa is not supported yet.
+## Online Upgrade (Version >= 5.18.0)
+
+### Online Upgrade Goals
+**1. Minimal downtime -** Ensure minimal downtime for each node.
+
+**2. Incremental changes -** Split upgrade so small chunks, for example, upgrade nodes one by one. Each node will get upgraded on its turn, the other nodes will still be available for handling s3 requests.
+
+**3. Rollback capability -** Mechanism for revert to the previous version in case something went wrong during the upgrade.
+
+**4. Schema backward compatibility -** Changes to account/bucket/config schema must be backwards compatible to allow seamless transitions to new version.
+
+
+### Online Upgrade Algorithm
+
+1. Initiate config directory backup (#1).
+2. Iterate nodes one by one -
+    * Stop NooBaa service (or suspend the node in CES)
+    * RPM upgrade each node.
+    * Restart NooBaa service on each node.
+3. Wait for all hosts to finish RPM upgrade (source code upgrade).
+4. Initiate config directory backup (#2).
+5. Initiate upgrade of the config directory using a noobaa-cli complete upgrade command. (point of no return)
+
+Online Upgrade Algorithm commands examples - 
+1. config directory backup -  
+    1. CES - `mms3 config backup /path/to/backup/location`
+    2. Non CES - `cp -R /etc/noobaa.conf.d/ /path/to/backup/location`
+2. Stop NooBaa service - `systemctl stop noobaa`
+3. RPM upgrade on a specific node - `rpm -Uvh /path/to/new_noobaa_rpm_version.rpm`
+4. Restart NooBaa service - `systemctl restart noobaa`
+5. `noobaa-cli upgrade start --expected_version=5.18.0 --expected_hosts=hostname1,hostname2,hostname3` 
+
+### Additional Upgrade Properties of `system.json`-
+
+1. New per host property -   
+    - config_dir_version
+
+2. New config directory information -
+    - config_directory 
+        - config_dir_version
+        - phase
+        - upgrade_package_version 
+        - in_progress_upgrade - (during the upgrade)
+            - timestamp
+            - completed_scripts
+            - running_host
+            - config_dir_from_version
+            - config_dir_to_version
+            - package_from_version
+            - package_to_version
+        - upgrade_history
+            - last_failure (if last upgrade failed)
+            - successful_upgrades
+
+#### system.json new information examples - 
+1. During Upgrade - `cat /etc/noobaa.conf.d/system.json | jq .`
```

Comment:
Could you add a line before/after each example with what the reader should notice?